src/Microsoft.DotNet.ServiceFabric.ServiceHost/LoggedServiceRemoting/AsyncInterceptor.cs

```diff
@@ -51,26 +56,42 @@ public virtual void Intercept(IInvocation invocation)
             }
             else if (IsTaskOfT(retType, out Type t))
             {
-                invocation.ReturnValue = s_interceptAsyncMethod.MakeGenericMethod(t)
-                    .Invoke(
-                        this,
-                        new[]
-                        {
-                            invocation,
-                            s_makeCallAsyncMethodMethod.MakeGenericMethod(t).Invoke(this, new object[] {invocation})
-                        });
+                try
+                {
+                    invocation.ReturnValue = s_interceptAsyncMethod.MakeGenericMethod(t)
+                        .Invoke(
+                            this,
+                            new[]
+                            {
+                                invocation,
+                                s_makeCallAsyncMethodMethod.MakeGenericMethod(t).Invoke(this, new object[] {invocation})
+                            });
+                }
+                catch (TargetInvocationException e) when (e.InnerException != null)
```

Comment:
Can we wrap the whole if statement chain with this catch, rather than just duplicating the catch?

Reply:
it doesn't technically apply to the first one, and I don't want to over-unwrap. If a "Task" thing thows a TargetInvocationException, I don't want to eat it.

Reply:
I can split the "else if" into "else try{ if " I suppose. :-)

Reply:
Yea, its fine the way it is. If we make that catch any bigger though we should consolidate.

---
In reply to: [428369141](https://github.com/dotnet/arcade-services/pull/1198#discussion_r428369141) [](ancestors = 428369141)